.github/workflows/build_ncs.yml

```diff
@@ -0,0 +1,64 @@
+name: Build binary for Nordic NCS
+
+# Runs when pushed to main, semver tagged release or any PR
+on:
+  push:
+    branches: [ main ]
+    tags: [ 'v*.*.*' ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    # Uses matrix to test combinations. Update when SDK changes or to add new boards.
+    strategy:
+      matrix:
+        ZEPHYR_SDK_VERSION: [0.16.1]
+        BOARD: ["nrf9160dk_nrf9160_ns","aludel_mini_v1_sparkfun9160_ns"]
+
+    steps:
+      - name: Install OS dependencies
+        run: >
+          sudo apt-get install -y
+          device-tree-compiler
+          git
+          ninja-build
+          python3
+          python3-pip
+          python3-wheel
+          wget
+          xz-utils
+
+      - name: Install build tools
+        run: |
+          pip3 install wheel
+          pip3 install cmake west
+
+      - name: Install Zephyr SDK
+        run: |
+          wget -q "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${{ matrix.ZEPHYR_SDK_VERSION }}/zephyr-sdk-${{ matrix.ZEPHYR_SDK_VERSION }}_linux-x86_64_minimal.tar.xz"
+          mkdir -p /opt/zephyr-sdk
+          tar -xvf zephyr-sdk-${{ matrix.ZEPHYR_SDK_VERSION }}_linux-x86_64_minimal.tar.xz -C /opt/zephyr-sdk --strip-components=1
+          /opt/zephyr-sdk/setup.sh -t arm-zephyr-eabi
+          rm zephyr-sdk-${{ matrix.ZEPHYR_SDK_VERSION }}_linux-x86_64_minimal.tar.xz
+
+      - name: Build with West
+        run: |
+          west init -m https://github.com/$GITHUB_REPOSITORY .
+          west update
+          west zephyr-export
+          pip3 install -r deps/zephyr/scripts/requirements.txt
+          west build -p -b ${{ matrix.BOARD }} app -- -DCONFIG_MCUBOOT_IMAGE_VERSION=\"0.0.0\"
+          mv build/zephyr/zephyr.bin build/zephyr/${{ matrix.BOARD }}.bin
```

Comment:
@szczys has been prefixing the manually generated binaries with the project name (e.g. https://github.com/golioth/reference-design-dc-power-monitor/releases/tag/v1.2.0). 

Should we do something similar here? e.g. is it possible to use the repo name as a prefix? i.e. something like:

`reference-design-air-quality_nrf9160dk_nrf9160_ns.bin`

Might also be nice if we could include the release version (and/or the git short rev):

```
reference-design-air-quality_v1.2.0_nrf9160dk_nrf9160_ns.bin          # version only
reference-design-air-quality_g0c85fbc_nrf9160dk_nrf9160_ns.bin        # git short rev only
reference-design-air-quality_v1.2.0-g0c85fbc_nrf9160dk_nrf9160_ns.bin # version & git short rev
```

Reply:
I was trying to use repo names so it would be dynamic but couldn't figure it out easily. I'll look into that further.

I think _if_ we want to continue to build for PRs & pushes to `main`, we'll need more complex logic for each event (ex. since a PR doesn't have a release.)

All do-able, just will take a bit more time :)

Reply:
For binaries in between releases the naming scheme is not bad maybe add the commit hash to the end of the file? There's a build artifact attached to the zip but I'm not sure how that can be traced back to the state of the repo when built.

Reply:
> There's a build artifact attached to the zip but I'm not sure how that can be traced back to the state of the repo when built.

Agreed, I think we really want the git rev as the source of truth for this.

> maybe add the commit hash to the end of the file

like this?

```
<repo-name>_<version>_<board-name>_<git-hash>.[bin, hex]
reference-design-air-quality_nrf9160dk_nrf9160_ns_g0c85fbc.bin        # git short rev only
reference-design-air-quality_v1.2.0_nrf9160dk_nrf9160_ns_g0c85fbc.bin # version & git short rev
```

other option:

```
<repo-name>_<board-name>_<version>-<git-hash>.[bin, hex]
reference-design-air-quality_nrf9160dk_nrf9160_ns_g0c85fbc.bin        # git short rev only
reference-design-air-quality_nrf9160dk_nrf9160_ns_v1.2.0-g0c85fbc.bin # version & git short rev
```

For comparison, the original scheme above is:

```
<repo-name>_<version>-<git-hash>_<board-name>.[bin, hex]
reference-design-air-quality_g0c85fbc_nrf9160dk_nrf9160_ns.bin        # git short rev only
reference-design-air-quality_v1.2.0-g0c85fbc_nrf9160dk_nrf9160_ns.bin # version & git short rev
```

Reply:
I went with `golioth-{{repo_name}}-{{board_name}}_{{file}}_{{short_git_hash}}.*`